sentry_sdk/metrics.py

```diff
@@ -495,8 +517,10 @@ def _get_aggregator_and_update_tags(key, tags):
 
     callback = client.options.get("_experiments", {}).get("before_emit_metric")
     if callback is not None:
-        if not callback(key, updated_tags):
-            return None, updated_tags
+        with recursion_protection() as in_metrics:
```

Comment:
Motivation here is to intentionally protect the callback. That way we only poke around on the thread local when we need to. This also means any user of the `_get_aggregator_and_update_tags` gets automatic protection and does not need to manually take care of marking their functions as noop.